### PR TITLE
Fix indent guide styling

### LINKED ISSUE #3298: Indent guides displaying without theming on blank lines
### Summary

On blank lines, indentation guides __other than the leftmost__ will display as white instead of their correct theme colour.
image
I tested with different themes, and oddly, it seems that it happens on most, but not all of the themes that come with Helix. Themes that seem to work fine:
* The default theme
* `base16_transparent`
* All the `catppuccin_*` themes
* `flatwhite`
* `meliora`
* `noctis` and `noctis_bordo`
* `nord_light`
* `onedark` and `onelight`
* `pop-dark` (side note: shouldn't this theme name use an underscore?)
* `solarized_dark` and `solarized_light`

In the case of the `base16_default` and `dracula_at_night` themes, the default indent guide colour seems to be white, so I don't know if this bug occurs, but for all the rest of the not listed themes, this bug does occur.

### Reproduction Steps

1. Use a theme not listed above.
2. Write code with at least 2 levels of indentation.
3. Insert a blank line somewhere at the second indentation level.
4. Ensure the blank line contains no whitespace characters.

### Helix log

```
2022-08-02T02:48:59.327 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:48:59.550 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:48:59.797 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:49:01.434 helix_view::theme [WARN] Theme: malformed hexcode: greay
2022-08-02T02:49:01.435 helix_view::theme [WARN] Theme: malformed hexcode: purple
2022-08-02T02:49:01.435 helix_view::theme [WARN] Theme: malformed hexcode: light-grey
2022-08-02T02:53:29.576 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:53:29.987 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:53:30.530 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:53:36.688 helix_view::theme [WARN] Theme: malformed hexcode: purple
2022-08-02T02:53:36.688 helix_view::theme [WARN] Theme: malformed hexcode: light-grey
2022-08-02T02:53:36.689 helix_view::theme [WARN] Theme: malformed hexcode: greay
2022-08-02T02:53:42.710 helix_view::theme [WARN] Theme: malformed hexcode: light-grey
2022-08-02T02:53:42.711 helix_view::theme [WARN] Theme: malformed hexcode: purple
2022-08-02T02:53:42.711 helix_view::theme [WARN] Theme: malformed hexcode: greay
2022-08-02T02:54:03.626 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:54:03.758 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:54:04.429 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:54:06.270 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:54:06.593 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:54:07.036 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:54:10.017 helix_view::theme [WARN] Theme: malformed hexcode: greay
2022-08-02T02:54:10.018 helix_view::theme [WARN] Theme: malformed hexcode: purple
2022-08-02T02:54:10.018 helix_view::theme [WARN] Theme: malformed hexcode: light-grey
2022-08-02T02:54:12.305 helix_view::theme [WARN] Theme: malformed hexcode: light-grey
2022-08-02T02:54:12.305 helix_view::theme [WARN] Theme: malformed hexcode: greay
2022-08-02T02:54:12.305 helix_view::theme [WARN] Theme: malformed hexcode: purple
2022-08-02T02:54:34.498 helix_view::theme [WARN] Theme: malformed hexcode: greay
2022-08-02T02:54:34.499 helix_view::theme [WARN] Theme: malformed hexcode: purple
2022-08-02T02:54:34.499 helix_view::theme [WARN] Theme: malformed hexcode: light-grey
2022-08-02T02:54:35.348 helix_view::theme [WARN] Theme: malformed hexcode: purple
2022-08-02T02:54:35.348 helix_view::theme [WARN] Theme: malformed hexcode: greay
2022-08-02T02:54:35.348 helix_view::theme [WARN] Theme: malformed hexcode: light-grey
2022-08-02T02:54:36.720 helix_view::theme [WARN] Theme: malformed hexcode: purple
2022-08-02T02:54:36.720 helix_view::theme [WARN] Theme: malformed hexcode: greay
2022-08-02T02:54:36.720 helix_view::theme [WARN] Theme: malformed hexcode: light-grey
2022-08-02T02:54:41.052 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:54:41.263 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:54:41.471 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:54:43.641 helix_view::theme [WARN] Theme: malformed hexcode: light-grey
2022-08-02T02:54:43.643 helix_view::theme [WARN] Theme: malformed hexcode: greay
2022-08-02T02:54:43.643 helix_view::theme [WARN] Theme: malformed hexcode: purple
2022-08-02T02:55:00.452 helix_view::theme [WARN] Theme: malformed hexcode: grey
2022-08-02T02:59:29.048 helix_view::theme [WARN] Theme: malformed hexcode: grey
```
This has led me to notice that the `base16_transparent` theme has a typo of `gray` as `greay`.

### Platform

MacOS

### Terminal Emulator

iTerm2 v3.4.16

### Helix Version

Helix 22.05 (`aa4394ce`)

### PR DESCRIPTION
**Pull request by A-Walrus**
Resolves #3298.
The problem was that the indent guides would render on top of whitespace, and get their theme. The incorrect looking white indent guides were actually correct according to the theme.
The first part of the fix is to make the indent guides actually use the indent guide theme.
The second part is to make indent guides a sub theme of whitespace so that when not defined they will look like whitespace, which is the desired behavior. I updated all themes that set `ui.virtual.indent-guide` to use `ui.virtual.whitespace.indent-guide`, as well as updating the documentation.

### CODE REVIEW

**Comment by the-mikedavis** [comment]
Instead of moving `indent-guides` under `ui.virtual.whitespace`, we could cheat a little and have a special fallback:

```rs
let indent_guide_style = theme.try_get("ui.virtual.indent-guide").unwrap_or_else(|| theme.get("ui.virtual.whitespace))
```

File: runtime/themes/dark_plus.toml [hunk 1]
```diff
@@ -81,7 +81,6 @@
 
 "ui.virtual.whitespace" = { fg = "dark_gray" }
 "ui.virtual.ruler" = { bg = "borders" }
-"ui.virtual.indent-guide" = { bg = "dark_gray4" }
```
Comment (the-mikedavis) [inline runtime/themes/dark_plus.toml:84]: This looks like an unrelated change? Although it does look better. Should it be

```toml
"ui.virtual.indent-guide" = { fg = "dark_gray4" }
```

instead?

**Review by the-mikedavis — APPROVED** [review]
Looks great!

I love a fix that makes the code simpler :)

File: helix-term/src/ui/editor.rs [hunk 2]
```diff
@@ -430,7 +434,7 @@ impl EditorView {
                     viewport.x + (i * tab_width as u16) - offset.col as u16,
                     viewport.y + line,
                     &indent_guide_char,
-                    indent_style,
+                    text_style.patch(indent_guide_style),
```
Comment (pickfire) [inline helix-term/src/ui/editor.rs:437]: Is it a good idea to patch `ui.virtual.indent-guide` if it exists? I think it should only patch `ui.virtual.whitespace` as a fallback rather than patching `ui.virtual.indent-guide` if it exists. I am still undecided since users may put `ui.virtual.indent-guide` without specifying a background.

Also, rather than doing this in a loop, I think it might be better if we patch it directly to `indent_guide_style` rather than doing this in the inner function.

Comment (A-Walrus) [reply helix-term/src/ui/editor.rs:437]: The reason I'm patching it with `text_style` is that if users only specify the foreground, it should use the background from `text_style`. If you look at line 457, all styles are patched from `text_style`
```rust
let style = spans
    .iter()
    .fold(text_style, |acc, span| acc.patch(theme.highlight(span.0)));
```
You're right about doing it outside of the loop, will fix.
